Failure with OCaml 4.00.x and 4.01.x

`setup-ocaml` now fails when `ocaml-compiler` is `4.00.x` or `4.01.x`:
- https://github.com/thierry-martinez/stdcompat/runs/7952976688?check_suite_focus=true
- https://github.com/thierry-martinez/stdcompat/runs/7952976827?check_suite_focus=true

The error is the following:
```
  Processing 13/21: [ocaml-base-compiler.4.01.0/bd7fa181cb64742c3b6cbb8ee13436554eb18cd7...fix-clang-build.diff: dl]
  [ERROR] Failed to get extra source "bd7fa181cb64742c3b6cbb8ee13436554eb18cd7...fix-clang-build.diff" of ocaml-base-compiler.4.01.0: curl error code 404
  
  #=== ERROR while fetching sources for ocaml-base-compiler.4.01.0 ==============#
  OpamSolution.Fetch_fail("https://github.com/jeremiedimino/ocaml/compare/bd7fa181cb64742c3b6cbb8ee13436554eb18cd7...fix-clang-build.diff (curl: code 404 while downloading https://github.com/jeremiedimino/ocaml/compare/bd7fa181cb64742c3b6cbb8ee13436554eb18cd7...fix-clang-build.diff)")
```

It used to work 10 days ago:
- https://github.com/thierry-martinez/stdcompat/runs/7802927146?check_suite_focus=true
- https://github.com/thierry-martinez/stdcompat/runs/7802927226?check_suite_focus=true

Moreover, `setup-ocaml` doesn't work with older versions of OCaml (<= 3.12), but this is not new as far as I can tell.
- With 3.07, it fails with the message `Error: No OCaml base compiler packages matched the version 3.07 in the opam-repository.`, whereas `opam switch create 3.07` should work as expected (https://github.com/thierry-martinez/stdcompat/runs/7952973003?check_suite_focus=true).
- With OCaml 3.08 to 3.12, it seems to fail because it tries to install `opam-depext` which requires OCaml >= 4.00.

--- a/packages/graphics/graphics.4.00.1/opam
+++ b/packages/graphics/graphics.4.00.1/opam
@@ -33,6 +33,6 @@ url {
 patches: "bd7fa181cb64742c3b6cbb8ee13436554eb18cd7...fix-clang-build.diff"
 extra-source "bd7fa181cb64742c3b6cbb8ee13436554eb18cd7...fix-clang-build.diff" {
   src:
-    "https://github.com/jeremiedimino/ocaml/compare/bd7fa181cb64742c3b6cbb8ee13436554eb18cd7...fix-clang-build.diff"
+    "https://raw.githubusercontent.com/ocaml/opam-source-archives/main/patches/fix-clang-build-ocaml-401.patch"
   checksum: "md5=faccda3b3ab092fa9ac7d5d4d8beb004"
 }

--- a/packages/graphics/graphics.4.01.0/opam
+++ b/packages/graphics/graphics.4.01.0/opam
@@ -33,6 +33,6 @@ url {
 patches: "bd7fa181cb64742c3b6cbb8ee13436554eb18cd7...fix-clang-build.diff"
 extra-source "bd7fa181cb64742c3b6cbb8ee13436554eb18cd7...fix-clang-build.diff" {
   src:
-    "https://github.com/jeremiedimino/ocaml/compare/bd7fa181cb64742c3b6cbb8ee13436554eb18cd7...fix-clang-build.diff"
+    "https://raw.githubusercontent.com/ocaml/opam-source-archives/main/patches/fix-clang-build-ocaml-401.patch"
   checksum: "md5=faccda3b3ab092fa9ac7d5d4d8beb004"
 }

--- a/packages/ocaml-base-compiler/ocaml-base-compiler.4.00.1/opam
+++ b/packages/ocaml-base-compiler/ocaml-base-compiler.4.00.1/opam
@@ -39,7 +39,7 @@ url {
 }
 extra-source "bd7fa181cb64742c3b6cbb8ee13436554eb18cd7...fix-clang-build.diff" {
   src:
-    "https://github.com/jeremiedimino/ocaml/compare/bd7fa181cb64742c3b6cbb8ee13436554eb18cd7...fix-clang-build.diff"
+    "https://raw.githubusercontent.com/ocaml/opam-source-archives/main/patches/fix-clang-build-ocaml-401.patch"
   checksum: "md5=faccda3b3ab092fa9ac7d5d4d8beb004"
 }
 extra-files: ["fix-gcc10.patch" "md5=eb8555e57846757138861e14d2665ed1"]

--- a/packages/ocaml-base-compiler/ocaml-base-compiler.4.01.0/opam
+++ b/packages/ocaml-base-compiler/ocaml-base-compiler.4.01.0/opam
@@ -51,7 +51,7 @@ url {
 }
 extra-source "bd7fa181cb64742c3b6cbb8ee13436554eb18cd7...fix-clang-build.diff" {
   src:
-    "https://github.com/jeremiedimino/ocaml/compare/bd7fa181cb64742c3b6cbb8ee13436554eb18cd7...fix-clang-build.diff"
+    "https://raw.githubusercontent.com/ocaml/opam-source-archives/main/patches/fix-clang-build-ocaml-401.patch"
   checksum: "md5=faccda3b3ab092fa9ac7d5d4d8beb004"
 }
 extra-files: ["fix-gcc10.patch" "md5=a5ee5f90499987223ca8c13896fa2c4c"]

--- a/packages/ocaml-variants/ocaml-variants.4.00.1+debug-runtime/opam
+++ b/packages/ocaml-variants/ocaml-variants.4.00.1+debug-runtime/opam
@@ -45,7 +45,7 @@ url {
 }
 extra-source "bd7fa181cb64742c3b6cbb8ee13436554eb18cd7...fix-clang-build.diff" {
   src:
-    "https://github.com/jeremiedimino/ocaml/compare/bd7fa181cb64742c3b6cbb8ee13436554eb18cd7...fix-clang-build.diff"
+    "https://raw.githubusercontent.com/ocaml/opam-source-archives/main/patches/fix-clang-build-ocaml-401.patch"
   checksum: "md5=faccda3b3ab092fa9ac7d5d4d8beb004"
 }
 available: !(os = "macos" & arch = "arm64")

--- a/packages/ocaml-variants/ocaml-variants.4.01.0+32bit/opam
+++ b/packages/ocaml-variants/ocaml-variants.4.01.0+32bit/opam
@@ -49,7 +49,7 @@ url {
 }
 extra-source "bd7fa181cb64742c3b6cbb8ee13436554eb18cd7...fix-clang-build.diff" {
   src:
-    "https://github.com/jeremiedimino/ocaml/compare/bd7fa181cb64742c3b6cbb8ee13436554eb18cd7...fix-clang-build.diff"
+    "https://raw.githubusercontent.com/ocaml/opam-source-archives/main/patches/fix-clang-build-ocaml-401.patch"
   checksum: "md5=faccda3b3ab092fa9ac7d5d4d8beb004"
 }
 available: !(os = "macos" & arch = "arm64")

--- a/packages/ocaml-variants/ocaml-variants.4.01.0+BER/opam
+++ b/packages/ocaml-variants/ocaml-variants.4.01.0+BER/opam
@@ -54,7 +54,7 @@ url {
 }
 extra-source "bd7fa181cb64742c3b6cbb8ee13436554eb18cd7...fix-clang-build.diff" {
   src:
-    "https://github.com/jeremiedimino/ocaml/compare/bd7fa181cb64742c3b6cbb8ee13436554eb18cd7...fix-clang-build.diff"
+    "https://raw.githubusercontent.com/ocaml/opam-source-archives/main/patches/fix-clang-build-ocaml-401.patch"
   checksum: "md5=faccda3b3ab092fa9ac7d5d4d8beb004"
 }
 available: !(os = "macos" & arch = "arm64")

--- a/packages/ocaml-variants/ocaml-variants.4.01.0+PIC/opam
+++ b/packages/ocaml-variants/ocaml-variants.4.01.0+PIC/opam
@@ -43,7 +43,7 @@ url {
 }
 extra-source "bd7fa181cb64742c3b6cbb8ee13436554eb18cd7...fix-clang-build.diff" {
   src:
-    "https://github.com/jeremiedimino/ocaml/compare/bd7fa181cb64742c3b6cbb8ee13436554eb18cd7...fix-clang-build.diff"
+    "https://raw.githubusercontent.com/ocaml/opam-source-archives/main/patches/fix-clang-build-ocaml-401.patch"
   checksum: "md5=faccda3b3ab092fa9ac7d5d4d8beb004"
 }
 available: !(os = "macos" & arch = "arm64")

--- a/packages/ocaml-variants/ocaml-variants.4.01.0+fp/opam
+++ b/packages/ocaml-variants/ocaml-variants.4.01.0+fp/opam
@@ -51,7 +51,7 @@ url {
 }
 extra-source "bd7fa181cb64742c3b6cbb8ee13436554eb18cd7...fix-clang-build.diff" {
   src:
-    "https://github.com/jeremiedimino/ocaml/compare/bd7fa181cb64742c3b6cbb8ee13436554eb18cd7...fix-clang-build.diff"
+    "https://raw.githubusercontent.com/ocaml/opam-source-archives/main/patches/fix-clang-build-ocaml-401.patch"
   checksum: "md5=faccda3b3ab092fa9ac7d5d4d8beb004"
 }
 available: !(os = "macos" & arch = "arm64")

--- a/packages/ocaml-variants/ocaml-variants.4.01.0+open-types/opam
+++ b/packages/ocaml-variants/ocaml-variants.4.01.0+open-types/opam
@@ -33,7 +33,7 @@ url {
 }
 extra-source "bd7fa181cb64742c3b6cbb8ee13436554eb18cd7...fix-clang-build.diff" {
   src:
-    "https://github.com/jeremiedimino/ocaml/compare/bd7fa181cb64742c3b6cbb8ee13436554eb18cd7...fix-clang-build.diff"
+    "https://raw.githubusercontent.com/ocaml/opam-source-archives/main/patches/fix-clang-build-ocaml-401.patch"
   checksum: "md5=faccda3b3ab092fa9ac7d5d4d8beb004"
 }
 available: !(os = "macos" & arch = "arm64")

--- a/packages/ocaml-variants/ocaml-variants.4.01.0+profile/opam
+++ b/packages/ocaml-variants/ocaml-variants.4.01.0+profile/opam
@@ -48,7 +48,7 @@ url {
 }
 extra-source "bd7fa181cb64742c3b6cbb8ee13436554eb18cd7...fix-clang-build.diff" {
   src:
-    "https://github.com/jeremiedimino/ocaml/compare/bd7fa181cb64742c3b6cbb8ee13436554eb18cd7...fix-clang-build.diff"
+    "https://raw.githubusercontent.com/ocaml/opam-source-archives/main/patches/fix-clang-build-ocaml-401.patch"
   checksum: "md5=faccda3b3ab092fa9ac7d5d4d8beb004"
 }
 available: !(os = "macos" & arch = "arm64")

